### PR TITLE
send params with spark request to train models

### PR DESCRIPTION
**Pull request by vansika**
# Problem
params used in training the model are picked from config making it less flexible for the developer.

# Solution
send params with spark request

### CODE REVIEW

File: listenbrainz/spark/request_manage.py [hunk 1]
```diff
@@ -137,11 +137,26 @@ def request_dataframes(days):
     send_request_to_spark_cluster(_prepare_query_message('cf_recording.recommendations.create_dataframes', params=params))
 
 
+def parse_list(ctx, args):
+    return list(args)
+
+
 @cli.command(name='request_model')
-def request_model():
+@click.option("--rank", callback=parse_list, default=[5, 10], type=int, multiple=True, help="Number of hidden features.")
```
Comment (mayhem) [inline listenbrainz/spark/request_manage.py:145]: These descriptions are not very useful for someone who doesn't really understand what is going on -- but, this is not the place to try and fix that. Would it be possible to print a link where the user can find more docs?

Comment (vansika) [reply listenbrainz/spark/request_manage.py:145]: https://spark.apache.org/docs/2.1.0/mllib-collaborative-filtering.html 

Looks good?

**Comment by vansika** [comment]
I think "help" is associated with a command. I am not sure if it will be a good practice to not show the link in help message of some commands.

**Comment by mayhem** [comment]
Can we show the link *once* when help for the request_model method is shown?

**Comment by vansika** [comment]
`request_model --help` will yield:

```
 Usage: manage.py spark request_model [OPTIONS]
  
Send the cluster a request to train the model. For more details refer to`
  'https://spark.apache.org/docs/2.1.0/mllib-collaborative-filtering.html'

Options:
  --rank INTEGER  Number of hidden features
  --itr INTEGER   Number of iterations to run.
  --lmbda FLOAT   Controls over fitting.
  --alpha FLOAT   Baseline level of confidence weighting applied.
  --help          Show this message and exit. 
```

**Comment by vansika** [comment]
and now I realise how important function docs can be :p

**Comment by vansika** [comment]
@paramsingh  We need [spark_manage.py](https://github.com/metabrainz/listenbrainz-server/blob/master/spark_manage.py) for easy debugging. The developers can easily invoke any script using spark_manage without having to send and receive requests.

But I feel there is a lot of code duplication in [spark_manage.py](https://github.com/metabrainz/listenbrainz-server/blob/master/spark_manage.py) and [reuest_manage](https://github.com/metabrainz/listenbrainz-server/blob/master/listenbrainz/spark/request_manage.py). I think it will good to merge them into one with options of sending a request via lemmy or directly invoking a spark script.

**Comment by paramsingh** [comment]
Can we move the commands in spark_manage.py into a manage.py subcommand instead?

**Comment by mayhem** [comment]
I'm happy with the link in the usage statement -- I'll leave you to work with Param on the manage.py business.

**Comment by vansika** [comment]
@paramsingh I am not sure how manage.py will solve the duplication issue. Can you be a little more descriptive? 

The commands and corresponding options in request_manage and spark_manage are identical. I was thinking of some flag for sending the requests via lemmy or directly?

**Comment by vansika** [comment]
@paramsingh This PR is ready to be merged.

**Pull request merged**